src/v/cluster/cluster_utils.h

```diff
@@ -25,6 +25,23 @@
 
 #include <utility>
 
+namespace detail {
+
+template<typename T, typename Fn>
+std::vector<cluster::topic_result>
+create_topic_results(const std::vector<T>& topics, Fn fn) {
```

Comment:
maybe use `Fn&&` ?

Reply:
I had that originally, Noah mentioned I didn't need that. https://github.com/vectorizedio/redpanda/pull/3264#discussion_r769155136

I'm actually a bit rusty on the details here, within a 5 minute google search I didn't find anything concrete. Thoughts?

Reply:
agreed, for those use cases we do not need forwarding reference